tasks/eva_2389/vcf_vertical_concat.py

```diff
@@ -0,0 +1,80 @@
+# Copyright 2021 EMBL - European Bioinformatics Institute
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Copyright 2021 EMBL - European Bioinformatics Institute
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import os
+from ebi_eva_common_pyutils.command_utils import run_command_with_output
+from ebi_eva_common_pyutils.logger import logging_config
+
+logger = logging_config.get_logger(__name__)
+logging_config.add_stdout_handler()
+
+
+class VerticalConcatProcess:
+    def __init__(self, files_to_concat_list: str, concat_processing_dir: str, output_vcf_file: str,
+                 bcftools_binary: str = "bcftools"):
+        self.files_to_concat_list = files_to_concat_list
+        self.concat_processing_dir = concat_processing_dir
+        self.output_vcf_file = output_vcf_file
+        self.bcftools_binary = bcftools_binary
+
+    def vertical_concat(self):
+        os.makedirs(os.path.dirname(self.output_vcf_file), exist_ok=True)
+        run_command_with_output(f"Running bcftools concat with the file list {self.files_to_concat_list}...",
+                                f"{self.bcftools_binary} concat "
+                                "--allow-overlaps --remove-duplicates "
+                                f"--file-list {self.files_to_concat_list} -o {self.output_vcf_file} -O z")
+        run_command_with_output(f"Running tabix on the output VCF {self.output_vcf_file}...",
+                                f"{self.bcftools_binary} index --csi {self.output_vcf_file}")
```

Comment:
Should this be `--tbi` or should the comment be modified?

Reply:
No, I explicitly put in CSI because [CSI indices support longer genomes](http://www.htslib.org/doc/tabix.html)

Reply:
Sure, I guess the comment could be changed then to match (not a big deal though).

Reply:
Addressed!